warehouse/forklift/legacy.py

```diff
@@ -767,6 +768,21 @@ def _is_duplicate_file(db_session, filename, hashes):
     return None
 
 
+def extract_wheel_metadata(path):
+    """
+    Extract METADATA file and return it as a content. The name of the
+    .whl file is used to find the corresponding .dist-info dir.
+
+    See https://www.python.org/dev/peps/pep-0658/#specification
+    """
+    global _wheel_file_re
```

Comment:
@merwok Python allows all kinds of bad programming tricks, such as monkey-patching, but that's no reason to use them. Global variables in programming [are considered to be bad practice](https://www.baeldung.com/cs/global-variables), so I am just marking their usage here.

Reply:
I'm not sure I get, because, then, should not you mark like that *every* object/name (imported module, defined class, etc.., really anything)  that you use in any function, following that logic ? but that would obviously not be practical..

and btw: using `global` in function is effectively considered "bad" practice. I mean : if you can do without/don't need it then it's really better to do without it.

Reply:
I agree, putting `global` there is not just a cosmetic change, it alters the usage.

Reply:
> It is a global access (just like referencing a module or class!) but not a mutation, so global is useless and unwanted.

Ok. I got your point. Without the `global`, the accidental assignment / modification of regex variable only affects the function code (global becomes shadowed), and with `global` it is possible to make modification that will ruin some other part of code where the global var is used.